runtimes/ruby-3.1/server.rb

```diff
@@ -3,117 +3,271 @@
 
 USER_CODE_PATH = '/usr/code-start';
 
-before do
-  content_type :json
-end
+class Response
+  def send(body, statusCode = 200, headers = {})
+    return {
+      'body' => body,
+      'statusCode' => statusCode,
+      'headers' => headers
+    }
+  end
 
-class RuntimeRequest
-  def initialize(payload = '{}', variables = {}, headers = {})
-    if payload == nil
-      payload = ''
-    end
+  def json(obj, statusCode = 200, headers = {})
+    headers['content-type'] = 'application/json'
+    return self.send(obj.to_json, statusCode, headers)
+  end
 
-    if headers == nil
-      headers = {}
-    end
+  def empty()
+    return self.send('', 204, {})
+  end
 
-    if variables == nil
-      variables = {}
-    end
+  def redirect(url, statusCode = 301, headers = {})
+    headers['location'] = url
+    return self.send('', statusCode, headers)
+  end
+end
 
-    @payload = payload
-    @variables = variables
+class Request
+  def initialize(bodyString, body, headers, method, url, path, port, scheme, host, query, queryString)
+    @bodyString = bodyString
+    @body = body
     @headers = headers
+    @method = method
+    @url = url
+    @path = path
+    @port = port
+    @scheme = scheme
+    @host = host
+    @query = query
+    @queryString = queryString
   end
 
-  def payload
-    @payload
+  attr_accessor :bodyString
+  attr_accessor :body
+  attr_accessor :headers
+  attr_accessor :method
+  attr_accessor :url
+  attr_accessor :path
+  attr_accessor :port
+  attr_accessor :scheme
+  attr_accessor :host
+  attr_accessor :query
+  attr_accessor :queryString
+end
+
+class Context
+  def initialize(rq, rs)
+    @req = rq
+    @res = rs
+    @logs = []
+    @errors = []
   end
 
-  def variables
-    @variables
+  attr_accessor :req
+  attr_accessor :res
+  attr_accessor :logs
+  attr_accessor :errors
+
+  def log(message)
+    if(message.kind_of?(Array) || message.kind_of?(Hash))
```

Comment:
It would be more Ruby-esque to omit the braces for `if` statements:

```suggestion
    if message.kind_of?(Array) || message.kind_of?(Hash)
```

Reply:
Updated everywhere